layers/+frameworks/vue/funcs.el

```diff
@@ -0,0 +1,135 @@
+  (setq imenu-generic-expression '(("html" "^<template>$" 0)
+                                   ("js" "^<script>$" 0)
+                                   ("js" "^\\s-*\\(data\\).*()\\s-?{" 1)
+                                   ("js" "^\\s-*\\(mounted\\).*()\\s-?{" 1)
+                                   ("js" "^\\s-*\\(beforeMount\\).*()\\s-?{" 1)
+                                   ("js" "^\\s-*\\(beforeDestroy\\).*()\\s-?{" 1)
+                                   ("js" "^\\s-*\\(created\\).*()\\s-?{" 1)
+                                   ("js" "^\\s-*\\(computed\\):\\s-?{" 1)
+                                   ("js" "^\\s-*\\(watch\\):\\s-?{" 1)
+                                   ("js" "^\\s-*\\(methods\\):\\s-?{" 1)
+                                   ("js" "^\\s-*\\(props\\):\\s-?{" 1)
+                                   ("css" "^<css>$" 0))
+        imenu-create-index-function #'imenu-default-create-index-function))
+
+(defun spacemacs//vue-setup-dumb ()
+  (add-to-list 'spacemacs-jump-handlers-vue-mode 'dumb-jump-go)
+  (spacemacs//vue-setup-dumb-imenu))
+
+(defun spacemacs//vue-setup-dumb-company ()
+  (spacemacs|add-company-backends :backends (company-web-html company-css company-files company-dabbrev)
+                                  :modes vue-mode
+                                  :variables company-minimum-prefix-length 2)
+  (company-mode))
+
+
+;; Others
+(defun spacemacs//vue-setup-yasnippet ()
+  (spacemacs/load-yasnippet)
+  (yas-activate-extra-mode 'js-mode))
+
+(defun spacemacs//vue-setup-editor-style ()
+  "such as indent rules comment style etc"
+  ;; https://stackoverflow.com/questions/36701024/how-can-i-indent-inline-javascript-in-web-mode
+  (setq web-mode-script-padding 0)
+  ;; https://emacs.stackexchange.com/questions/27683/change-comment-style-in-web-mode
+  (add-to-list 'web-mode-comment-formats '("javascript" . "//")))
+
+(defun spacemacs//vue-setup-keybindings ()
+  "free stuff from `html' layer"
+  (spacemacs/declare-prefix-for-mode 'vue-mode "mE" "errors")
+  (spacemacs/declare-prefix-for-mode 'vue-mode "me" "element")
+  (unless (equal vue-backend 'lsp) 
+    (spacemacs/declare-prefix-for-mode 'vue-mode "mh" "help")
+    (spacemacs/declare-prefix-for-mode 'vue-mode "r" "refactor"))
+  (spacemacs/set-leader-keys-for-major-mode 'vue-mode
+    "El" 'web-mode-dom-errors-show
+    "eb" 'web-mode-element-beginning
+    "ec" 'web-mode-element-child
+    "ep" 'web-mode-element-parent
+    "es" 'web-mode-element-sibling-next
+    "hp" 'web-mode-dom-xpath
+    "rc" 'web-mode-element-clone
+    "rd" 'web-mode-element-vanish
+    "rk" 'web-mode-element-kill
+    "rn" 'web-mode-element-rename
+    "rw" 'web-mode-element-wrap
+    "z" 'web-mode-fold-or-unfold)
+  (spacemacs|define-transient-state vue-mode
+    :title "Web-mode Transient State"
+    :columns 4
+    :foreign-keys run
+    :evil-leader-for-mode (vue-mode . ".")
+    :bindings
+    ("j" web-mode-element-next "next")
```

Comment:
It might be nice to group the key bindings by the command name, so that for example the:
`web-mode-element-sibling-next` and `*-previous` key bindings are next to each other.

But if they are sorted alphabetically by the key sequence, then it's easier to:
- see which key bindings are available.
- and to prevent accidentally binding two different commands to the same key binding.

Reply:
Agree, I copied this from `html` layer. Not sure if we should do the same to it

Reply:
Ok, I see now, they seem to be ordered after how the keys are listed in the transient state.

When this layer is merged. Then I can fix the transient state formatting and backend key binding order/sorting in both the vue and html layers.